entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md

```diff
@@ -14,8 +14,69 @@ This page will document API and behavior changes that have the potential to brea
+| [`Contains` in LINQ queries may stop working on older SQL Server versions                                                                | High       |
 | [SQL Server `date` and `time` now scaffold to .NET `DateOnly` and `TimeOnly`](#sqlserver-date-time-only)                                 | Medium     |
 
+## High-impact changes
+
+<a name="sqlserver-contains-compatibility"></a>
+
+### `Contains` in LINQ queries may stop working on older SQL Server versions
+
+[Tracking Issue #13617](https://github.com/dotnet/efcore/issues/13617)
+
+#### Old behavior
+
+Previously, when the `Contains` operator was used in LINQ queries with a parameterized value list, EF generated SQL that was inefficient but worked on all SQL Server versions.
+
+#### New behavior
+
+Starting with EF Core 8.0, EF now generates SQL that is more efficient, but is unsupported on SQL Server 2014 and below.
+
+Note that newer SQL Server versions may be configured with an older [compatibility level](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16), also making them incompatible with the new SQL. This can also occur with an Azure SQL database which was migrated from a previous on-premises SQL Server instance, carrying over the old compatibility level.
+
+#### Why
+
+The previous SQL generated by EF Core for `Contains` inserted the parameterized values as constants in the SQL. For example, the following LINQ query:
+
+```c#
+var names = new[] { "Blog1", "Blog2" };
+
+var blogs = await context.Blogs
+    .Where(b => names.Contains(b.Name))
+    .ToArrayAsync();
+```
+
+... would be translated to the following SQL:
+
+```sql
+SELECT [b].[Id], [b].[Name]
+FROM [Blogs] AS [b]
+WHERE [b].[Name] IN (N'Blog1', N'Blog2')
+```
+
+Such insertion of constant values into the SQL creates many performance problems, defeating query plan caching and causing unneeded evictions of other queries. The new EF Core 8.0 translation uses the SQL Server [`OPENJSON`](https://learn.microsoft.com/sql/t-sql/functions/openjson-transact-sql) function to instead transfer the values as a JSON array. This solves the performance issues inherent in the previous technique; however, the `OPENJSON` function is unavailable in SQL Server 2014 or below.
+
+For more information about this change, [see this blog post](https://devblogs.microsoft.com/dotnet/announcing-ef8-preview-4/).
+
+#### Mitigations
+
+If your database is SQL Server 2016 (13.x) or newer, or if you're using Azure SQL, check the configured compatibility level of your database via the following command:
+
+```sql
+SELECT name, compatibility_level FROM sys.databases;
+```
+
+If the compatibility level is below 130 (SQL Server 2016), consider modifying it to a newer value ([documentation](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16)).
```

Comment:
https://learn.microsoft.com/sql/t-sql/statements/alter-database-transact-sql-compatibility-level

Reply:
Maybe link specifically to this? https://learn.microsoft.com/sql/t-sql/statements/alter-database-transact-sql-compatibility-level#best-practices-for-upgrading-database-compatibility-level